docs/source/installation.rst

```diff
@@ -2,19 +2,213 @@
 Installation
 ============
 
-BentoML is distributed as a Python package available `on PyPI <https://pypi.org/project/bentoml/>`_.
+Install 🍱 BentoML with your favourite package manager, alongside with whichever deep learning
+library you are working with, and you are ready to go!
+
+🍱 BentoML is distributed as a Python Package available `on PyPI <https://pypi.org/project/bentoml/>`_.
 
 * BentoML supports Linux/UNIX, Windows, and MacOS.
 * BentoML requires Python 3.7 or above.
 
+.. code-block::
+
+   pip install --user bentoml
+
+
+Install using virtual environment
+---------------------------------
+
+You should install BentoML in a `virtual environment <https://docs.python.org/3/library/venv.html>`_. If you are not familar with virtual environment, refers to this
+`guide <https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/>`_. Virtual environment are a great way to manage dependencies and
+avoid compatibility issues among different projects.
+
+Start create a virtual environment in your project directory
+
+.. code-block:: bash
+
+    python3 -m venv ./venv
+
+.. tip::
+
+   Alternatively, you can also use `virtualenv <https://virtualenv.pypa.io/en/latest/>`_ that provides more features than the built-in venv module.
+
+Activate the virtual environment:
+
+.. tab-set::
+
+    .. tab-item:: MacOS/UNIX
+
+        .. code-block:: bash
+
+            source ./venv/bin/activate
+
+    .. tab-item:: Windows
+
+        .. code-block:: powershell
+
+            .\venv\Scripts\activate
+
+Now, you can install BentoML via pip:
+
 .. code-block:: bash
 
     pip install bentoml
 
-You may need to install additional libraries to use certain BentoML modules or features.
-For example, the :code:`bentoml.tensorflow` module requires TensorFlow to be installed;
-The :code:`bentoml.io.Image` class requires :code:`Pillow` to be installed; and to
-export saved models to s3, the :code:`fs-s3fs` package is required.
+Additional features
```

Comment:
I still don't think "Additional Features" belongs in here. Think from a user's perspective, they come to the doc looking for installation instructions. At this time they have no intention or understanding of what installing additional feature mean. They will encounter this as they start using the features.

Reply:
I think that having an `Additional Features` section is helpful as our additional features are relatively complex.

I do understand the your viewpoint. Nonetheless I think this is still helpful for those who needs to install said features.

We can move it lower.

Reply:
The installation page isn't how users should discover about the additional features. If you feel strong about including them here, I'd recommend that we add a separate "advanced installation" page.

Reply:
Having this section into a separate `advanced installation` doesn't make sense.

mlflow is doing this but just listing all of the options they can install

transformers are also doing the same with their additional deps

spacy is also listing all of the features in their installation guides.

I don't think this is that big of an issue. It would rather we expose here than just reference to just a separate page.